pkg/client/client.go

```diff
@@ -54,13 +55,22 @@ func (e *ApiCallError) Is(target error) bool {
 }
 
 type AuthnOpts struct {
-	OIDCToken *OIDCToken
+	OIDCToken      *OIDCToken
+	HTTPBasicAuthn *HTTPBasicAuthnConfig
 }
 
 type OIDCToken struct {
 	Value string
 }
 
```

Comment:
move the `HTTPBasicAuthnConfig` and the logic to get the username value into the config code in the `cli` package, the client code should receive the username value keeping the separation between the configuration logic and the http client logic.

We could have something like the following in the client package.

```
type AuthnOpts struct {
	OIDCToken *OIDCToken
	HTTPBasic *HTTPBasic
}

type HTTPBasic struct {
  Username string
}
```